### PR TITLE
Fix X-Ray Service Map filter trace list query

### PR DESCRIPTION
**Pull request by jamesrwhite**
This fixes https://github.com/grafana/x-ray-datasource/issues/153

I have tested it on Grafana `8.4.11` and `10.1.2`.

I tried to use the `NodeGraphDataFrameFieldNames.subTitle` enum directly to generate the query but was struggling to get that to build correctly. If there's a better way to achieve this please point me in that direction and I'll happily change it, I'm aware this approach is pretty hacky 😅

### CODE REVIEW

File: src/DataSource.ts [hunk 1]
```diff
@@ -276,7 +277,12 @@ function parseServiceMapResponse(
   query?: XrayQuery
 ): DataFrame[] {
   const [servicesFrame, edgesFrame] = parseGraphResponse(response, query);
-  const serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subTitle}"))';
+  let serviceQuery;
+  if (NodeGraphDataFrameFieldNames.subTitle === 'subtitle') {
+    serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subtitle}"))';
+  } else {
+    serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subTitle}"))';
+  }
```
Comment (fridgepoet) [inline src/DataSource.ts:285]:
```suggestion
const serviceQuery = `service(id(name: "\${__data.fields.title}", type: "\${__data.fields.${NodeGraphDataFrameFieldNames.subTitle}}"))`;
```

(I think this also works)

**Review by fridgepoet — APPROVED** [review]
Hey @jamesrwhite thanks so much for the contribution and your notes on the original issue!

To reproduce: I queried the Service Map, switched to Node Graph visualization, and chose a node and selected Traces > All.

I tested out the changes locally on:

- v8.4.7 - OK
- v8.5.1 - OK
- v9.4.0 - OK
- latest main v10.3.0-pre - OK

I was able to see the subtitle variable transform correctly.

File: src/DataSource.ts (hunk 1 above)
Comment (jamesrwhite) [reply src/DataSource.ts:285]: This is definitely a cleaner approach and I tried something like this although I wasn't able to confirm this works across the different versions given my issues with getting the plugin to build.

Are you able to test it with this change?

Comment (jamesrwhite) [reply src/DataSource.ts]: Great, thanks for checking 🙏

**Comment by fridgepoet** [comment]
Hi again @jamesrwhite, I got some help from @idastambuk to write a test to describe what we're doing here. It's the last commit on this branch https://github.com/grafana/x-ray-datasource/tree/test-proposal
Would you be willing to incorporate that?

**Comment by jamesrwhite** [comment]
@fridgepoet Absolutely, thanks for that. I've just merged that branch into this one 🚀

**Review by sarahzinger — APPROVED** [review]
Awesome! Thanks so much for your contribution!

**Pull request merged**